File: frontend/src/services/userService.js
import api from '@/utils/api'

const userService = {
  register(username, email, password) {
    return api.post('/users/register', { username, email, password })
  },

  login(username, password) {
    return api.post('/users/login', { username, password })
  },

  getUserInfo(id) {
    return api.get(`/users/${id}`)
  },

  updateUserInfo(id, avatar, bio) {
    return api.put(`/users/${id}`, { avatar, bio })
  },

  uploadAvatar(file) {
    const formData = new FormData()
    formData.append('file', file)
    return api.post('/users/upload-avatar', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
  },

  getUserPoints(id) {
    return api.get(`/users/${id}/points`)
  },

  getPointsRanking(limit = 100) {
    return api.get('/users/ranking', {
      params: { limit }
    })
  }
}

export default userService

File: backend/src/main/java/com/qa/service/UserService.java
package com.qa.service;

import com.qa.dto.UserDTO;
import com.qa.entity.User;

/**
 * 用户业务服务
 */
public interface UserService {
    
    /**
     * 用户注册
     */
    User registerUser(String username, String email, String password);
    
    /**
     * 用户登录
     */
    String loginUser(String username, String password);
    
    /**
     * 获取用户信息
     */
    UserDTO getUserInfo(Long userId);
    
    /**
     * 更新用户信息
     */
    User updateUserInfo(Long userId, String avatar, String bio);
    
    /**
     * 获取用户积分
     */
    Integer getUserPoints(Long userId);
    
    /**
     * 更新用户积分
     */
    void updateUserPoints(Long userId, Integer points);
    
    /**
     * 获取积分排行榜
     */
    java.util.List<UserDTO> getPointsRanking(Integer limit);
    
    /**
     * 查询用户是否存在
     */
    boolean userExists(String username);
}
